### PR TITLE
Reactive model g379

### PR DESCRIPTION
**Pull request by oehm-smith**
Firstly note that I will do a PR to reformat after this.

There is a months work here.  All other devs changes have been merged in.

This change converts the template forms to reactive forms.

TODO:
* GEOD-370 turn off revert button when nothing has changed (simple)
* GEOD-396 ResponsibleParty styling needs fixing (simple)
* GEOD-408 siteDataCentre incorrect. Is it a responsibleParty? Or list in MoreInformation? (medium)
* GEOD-409 Add ViewModel Mapping for SiteInformation (Location, Identification) (medium)
* GEOD-410 Fix JsonDiff for new way we insert items (complex)
* GEOD-407 Fix Validators on Model driven forms (complex)

### CODE REVIEW

File: src/client/app/frequency-standard/frequency-standard-item.component.html [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
       </span>
-      <span *ngIf="index != 0">
-        <span class="hidden-xsm">Previous </span>
-        <span class="hidden-xxs">Frequency Standard (</span>
-        {{ miscUtils.getDate(frequencyStandard.startDate) }} &ndash;
-        {{ miscUtils.getDate(frequencyStandard.endDate) }}
-        <span class="hidden-xxs">)</span>
+        <span *ngIf="index != 0">
+        Previous Frequency Standard ({{ miscUtils.getDate(frequencyStandard.startDate) }}
+        &ndash; {{ miscUtils.getDate(frequencyStandard.endDate) }})
```
Comment (frankfu03) [inline src/client/app/frequency-standard/frequency-standard-item.component.html]: It is a simplified version to show current/previous items, but may cause problem to small mobile devices. I think this PR is to focus on adding reactive forms to UI.

File: src/client/app/site-info/site-info.component.html [hunk 2]
```diff
@@ -1,159 +1,77 @@
 <div class="container centered clear-pad">
   <div class="col-md-12 col-sm-12 col-xs-12 col-xxs-12 pad-sm">
-    <form #siteInfoForm="ngForm" novalidate class="form-horizontal">
-      <div class="panel-group">
+    <form [formGroup]="siteInfoForm" (ngSubmit)="newSave(siteInfoForm)" novalidate class="form-horizontal" >
+    <div class="panel-group">
+        <button type="submit" [disabled]="!siteInfoForm.dirty">SAVE</button><span>Dirty?: {{siteInfoForm.dirty}}</span>
```
Comment (frankfu03) [inline src/client/app/site-info/site-info.component.html]: Is it for testing?

File: src/client/app/frequency-standard/frequency-standard-item.component.html (hunk 1 above)
Comment (oehm-smith) [inline src/client/app/frequency-standard/frequency-standard-item.component.html]: Yes I didn't make that change (that I know of).  The merge must have gone a muck.  I'll restore the other code.

File: src/client/app/site-info/site-info.component.html (hunk 2 above)
Comment (oehm-smith) [inline src/client/app/site-info/site-info.component.html]: Yep, thanks.

**Pull request merged**